activerecord/lib/active_record/future_result.rb

```diff
@@ -5,6 +5,7 @@ class FutureResult # :nodoc:
     class Complete
       attr_reader :result
       delegate :empty?, :to_a, to: :result
+      delegate_missing_to :result
```

Comment:
If I comment this out your test doesn't fail.

Reply:
I'll double-check it

Reply:
This bug has already been fixed in https://github.com/rails/rails/pull/50778. There's no longer a need for `delegate_missing_to`

Reply:
Ok. I don't mind the extra tests then. If you squash your commits I'll merge.

Reply:
ok, thanks